### ISSUE TITLE
Error: "Parse failure: Unexpected token"

### ISSUE THREAD

**Issue by silveltman** [ACP-CODE/astro-favicons#3]
First of all, thanks for creating this integration!

however, I get the following error:
```
Parse failure: Unexpected token (24:184)
At file: /src/layouts/Layout.astro
Contents of line 24:   return $$render(_a || (_a = __template(['<html lang="nl" class="theme-dark size-medium mode-base hue-base bg-hue1 scroll-smooth" data-astro-cid-sckkx6r4><head>', "", "", "<link rel="icon" type="image/x-icon" href="/favicon.ico" media="(prefers-color-scheme: light)"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"  media="(prefers-color-scheme: light)"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"  media="(prefers-color-scheme: light)"><link rel="icon" type="image/png" sizes="48x48" href="/favicon-48x48.png"  media="(prefers-color-scheme: light)"><link rel="icon" type="image/svg+xml" href="/favicon.svg"><link rel="icon" type="image/x-icon" href="/favicon-dark.ico" media="(prefers-color-scheme: dark)"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16-dark.png"  media="(prefers-color-scheme: dark)"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32-dark.png"  media="(prefers-color-scheme: dark)"><link rel="icon" type="image/png" sizes="48x48" href="/favicon-48x48-dark.png"  media="(prefers-color-scheme: dark)"><link rel="manifest" href="/manifest.webmanifest"><meta name="mobile-web-app-capable" content="yes"><meta name="theme-color" content="#fff"><meta name="application-name" content="My website"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-status-bar-style" content="black-translucent"><meta name="apple-mobile-web-app-title" content="Astro Favicons"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#fff"><meta name="msapplication-TileColor" content="#fff"><meta name="msapplication-config" content="/browserconfig.xml"><link rel="yandex-tableau-widget" href="/yandex-browser-manifest.json"></head><body data-astro-cid-sckkx6r4>", "", '<script defer src="https://cdn.jsdelivr.net/npm/@alpinejs/intersect@3.x.x/dist/cdn.min.js"><\/script><script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.x.x/dist/cdn.min.js"><\/script></body></html>'])), $$renderComponent($$result, "Head", Head, { "meta": meta, "data-astro-cid-sckkx6r4": true }), $$renderComponent($$result, "ViewTransitions", ViewTransitions, { "data-astro-cid-sckkx6r4": true }), $$renderHead($$result), $$renderSlot($$result, $$slots["default"]), $$renderComponent($$result, "LivePreview", null, { "client:only": true, "client:component-hydration": "only", "data-astro-cid-sckkx6r4": true, "client:component-path": "@components/LivePreview.svelte", "client:component-export": "default" }));

**Comment by JakiChen** [comment]
> First of all, thanks for creating this integration!
> 
> however, I get the following error:
> 
> ```
> Parse failure: Unexpected token (24:184)
> At file: /src/layouts/Layout.astro
> Contents of line 24:   return $$render(_a || (_a = __template(['<html lang="nl" class="theme-dark size-medium mode-base hue-base bg-hue1 scroll-smooth" data-astro-cid-sckkx6r4><head>', "", "", "<link rel="icon" type="image/x-icon" href="/favicon.ico" media="(prefers-color-scheme: light)"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"  media="(prefers-color-scheme: light)"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"  media="(prefers-color-scheme: light)"><link rel="icon" type="image/png" sizes="48x48" href="/favicon-48x48.png"  media="(prefers-color-scheme: light)"><link rel="icon" type="image/svg+xml" href="/favicon.svg"><link rel="icon" type="image/x-icon" href="/favicon-dark.ico" media="(prefers-color-scheme: dark)"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16-dark.png"  media="(prefers-color-scheme: dark)"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32-dark.png"  media="(prefers-color-scheme: dark)"><link rel="icon" type="image/png" sizes="48x48" href="/favicon-48x48-dark.png"  media="(prefers-color-scheme: dark)"><link rel="manifest" href="/manifest.webmanifest"><meta name="mobile-web-app-capable" content="yes"><meta name="theme-color" content="#fff"><meta name="application-name" content="My website"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-status-bar-style" content="black-translucent"><meta name="apple-mobile-web-app-title" content="Astro Favicons"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#fff"><meta name="msapplication-TileColor" content="#fff"><meta name="msapplication-config" content="/browserconfig.xml"><link rel="yandex-tableau-widget" href="/yandex-browser-manifest.json"></head><body data-astro-cid-sckkx6r4>", "", '<script defer src="https://cdn.jsdelivr.net/npm/@alpinejs/intersect@3.x.x/dist/cdn.min.js"><\/script><script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.x.x/dist/cdn.min.js"><\/script></body></html>'])), $$renderComponent($$result, "Head", Head, { "meta": meta, "data-astro-cid-sckkx6r4": true }), $$renderComponent($$result, "ViewTransitions", ViewTransitions, { "data-astro-cid-sckkx6r4": true }), $$renderHead($$result), $$renderSlot($$result, $$slots["default"]), $$renderComponent($$result, "LivePreview", null, { "client:only": true, "client:component-hydration": "only", "data-astro-cid-sckkx6r4": true, "client:component-path": "@components/LivePreview.svelte", "client:component-export": "default" }));
> ```

### Thank you for your attention and feedback

It’s not clear which version you are using. Based on your repositorie [fulldev-V2](https://github.com/silveltman/fulldev-V2), the bug has not been reproduced yet.

- **astro-favicons^1.1.0**

#### The logic of inserting favicon tags into each page is implemented by updating vite settings.

```js
export async function vitePluginFavicons(src: string, options: FaviconOptions, compressHTML: boolean) {

  const response = await favicons(src, options);
  let htmlTags: string;

  if (compressHTML) {
    htmlTags = `${response.html.join('').replaceAll('\n', '')}`;
  } else {
    htmlTags = `\n\n<!-- Astro Favicons v1.1.0 - https://github.com/ACP-CODE/astro-favicons -->\n${response.html.join('\n').replace(/(?<!\n)\n\n+(?!\n)/g, '\n')}\n<!--  Astro Favicons -->\n\t`;
  }

  return {
    name: 'vite-plugin-favicons',
    enforce: 'pre',
    transform(html: string) {
      try {
        return html.replace('</head>', `${htmlTags}</head>`);
      } catch (error) {
        throw error;
      }
    }
  }

};
```

#### The currently known bug is that an error will occur when the following code exists on a single layout page.

```astro
---
import BaseHead from "components/BaseHead/BaseHead.astro";

import type { Props as BaseHeadProps } from "components/BaseHead/BaseHead.astro";
export interface Props {
	Seo?: BaseHeadProps;
}

const { Seo } = Astro.props;
---
<!DOCTYPE html>
<html lang="en">
	<head>
		<BaseHead {...Seo} />
	</head>
	<body>
		<slot />
	</body>
</html>
<script is:inline>
  console.log("hello....")
</script>
```

 #### The solution is to remove `is:inline` in <script> tag or change the code. Embed other pages.

**Comment by RafidMuhymin** [comment]
@JakiChen I am facing the same issue with a `<script define:vars={}>` tag inside the head. I understand that this is a limitation but why is it? And will it get a fix in the future?

**Comment by JakiChen** [comment]
> First of all, thanks for creating this integration!
> 
> however, I get the following error:
> 
> ```
> Parse failure: Unexpected token (24:184)
> At file: /src/layouts/Layout.astro
> Contents of line 24:   return $$render(_a || (_a = __template(['<html lang="nl" class="theme-dark size-medium mode-base hue-base bg-hue1 scroll-smooth" data-astro-cid-sckkx6r4><head>', "", "", "<link rel="icon" type="image/x-icon" href="/favicon.ico" media="(prefers-color-scheme: light)"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png"  media="(prefers-color-scheme: light)"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png"  media="(prefers-color-scheme: light)"><link rel="icon" type="image/png" sizes="48x48" href="/favicon-48x48.png"  media="(prefers-color-scheme: light)"><link rel="icon" type="image/svg+xml" href="/favicon.svg"><link rel="icon" type="image/x-icon" href="/favicon-dark.ico" media="(prefers-color-scheme: dark)"><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16-dark.png"  media="(prefers-color-scheme: dark)"><link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32-dark.png"  media="(prefers-color-scheme: dark)"><link rel="icon" type="image/png" sizes="48x48" href="/favicon-48x48-dark.png"  media="(prefers-color-scheme: dark)"><link rel="manifest" href="/manifest.webmanifest"><meta name="mobile-web-app-capable" content="yes"><meta name="theme-color" content="#fff"><meta name="application-name" content="My website"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-status-bar-style" content="black-translucent"><meta name="apple-mobile-web-app-title" content="Astro Favicons"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#fff"><meta name="msapplication-TileColor" content="#fff"><meta name="msapplication-config" content="/browserconfig.xml"><link rel="yandex-tableau-widget" href="/yandex-browser-manifest.json"></head><body data-astro-cid-sckkx6r4>", "", '<script defer src="https://cdn.jsdelivr.net/npm/@alpinejs/intersect@3.x.x/dist/cdn.min.js"><\/script><script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.x.x/dist/cdn.min.js"><\/script></body></html>'])), $$renderComponent($$result, "Head", Head, { "meta": meta, "data-astro-cid-sckkx6r4": true }), $$renderComponent($$result, "ViewTransitions", ViewTransitions, { "data-astro-cid-sckkx6r4": true }), $$renderHead($$result), $$renderSlot($$result, $$slots["default"]), $$renderComponent($$result, "LivePreview", null, { "client:only": true, "client:component-hydration": "only", "data-astro-cid-sckkx6r4": true, "client:component-path": "@components/LivePreview.svelte", "client:component-export": "default" }));
> ```

> @JakiChen I am facing the same issue with a `<script define:vars={}>` tag inside the head. I understand that this is a limitation but why is it? And will it get a fix in the future?

@silveltman , @RafidMuhymin 

[issue#3](https://github.com/ACP-CODE/astro-favicons/issues/3) may be fixed in latest version **astro-favicons^1.2.0**

See the latest code [src/core.ts](https://github.com/ACP-CODE/astro-favicons/blob/0b83fce3d71b5fbc3fade464df38cacc75f1a1b2/src/core.ts#L170) for details
